### DISCUSSION TITLE
Spring Boot 1.x adapter deprecation

### DISCUSSION THREAD

**Discussion by mabartos** [keycloak/keycloak#9936] [Archive]
Keycloak is deprecating most adapters in the future, for more information see https://www.keycloak.org/2022/02/adapter-deprecation.html.

I'd like to ask if anybody still uses the Keycloak adapter for Spring Boot 1.x? The Keycloak codebase contains two separate adapters for Spring Boot, such as for versions 1.x and 2.x. The question is if it's still worth keeping it included in the codebase. Even Red Hat doesn't support Spring Boot version 1.5.x [1], which is the most recent version for 1.5.x (if I'm not mistaken). 

[1] https://access.redhat.com/articles/3349341

**Comment by nosskranitz** [comment]
Hi,

I see in our app that we are using 
import org.keycloak.adapters.springboot.KeycloakSpringBootConfigResolver
import org.keycloak.adapters.springsecurity.KeycloakConfiguration
import org.keycloak.adapters.springsecurity.authentication.KeycloakAuthenticationEntryPoint
import org.keycloak.adapters.springsecurity.config.KeycloakWebSecurityConfigurerAdapter
import org.keycloak.adapters.KeycloakDeployment
import org.keycloak.adapters.KeycloakDeploymentBuilder
import org.keycloak.adapters.spi.HttpFacade
import org.keycloak.adapters.springboot.KeycloakSpringBootConfigResolver
import org.keycloak.representations.adapters.config.AdapterConfig

from 
org.keycloak:keycloak-spring-security-adapter:15.0.0
org.keycloak:keycloak-spring-boot-adapter-core:15.0.0
org.keycloak:keycloak-adapter-core:15.0.0
org.keycloak:keycloak-adapter-spi:15.0.

Could you tell me if there will be an impact for us regarding keycloak spring decrecation https://www.keycloak.org/2022/02/adapter-deprecation ?

Regards,

**Reply by stianst** [reply]
We are deprecating all Java adapters, which includes Spring Boot/Security adapters. See https://github.com/keycloak/keycloak/discussions/10187.

This discussion is specifically for the older Spring Boot 1.x adapter, which we don't believe anyone is using and can most likely remove earlier than what is mentioned in the link above.

**Reply by nosskranitz** [reply]
Hi,

Thanks for answering. I am still not sure if this is concerning us ? We use springboot 2.6.7. Will the adapters be kept up to date for springboot 2.X ?

We are pulling the following dependencies :

        <dependency>
            <groupId>org.keycloak.bom</groupId>
            <artifactId>keycloak-adapter-bom</artifactId>
            <version>18.0.0</version>
            <type>pom</type>
        </dependency>
        <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-spring-boot-starter</artifactId>
            <version>18.0.0</version>
        </dependency>
        <dependency>
            <groupId>org.keycloak</groupId>
            <artifactId>keycloak-admin-client</artifactId>
            <version>18.0.0</version>
        </dependency>

Best regards,

**Reply by ahmedgazie** [reply]
@stianst Can you please check the above question? we are using spring boot 2.x... Will the adapters be kept up to date for springboot 2.X?

**Comment by stianst** [comment]
I'm in agreement with removing the legacy Spring Boot adapter already now, and we shouldn't have to wait for the deprecation dates we have in https://www.keycloak.org/2022/02/adapter-deprecation.html
